### PR TITLE
Querybuilder new features

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Add last insert ID (because of table prefix and quoting and ...)
- [x] Add `select distinct`
- [x] Make method to get quoted/prefixed table and column names public

cc @schiesbn as requested

Review @LukasReschke @MorrisJobke

### CODE REVIEW

File: lib/private/db/querybuilder/querybuilder.php [hunk 1]
```diff
@@ -1024,14 +1046,46 @@ public function createFunction($call) {
...
+	public function getTableName($table) {
 		if ($this->automaticTablePrefix === false || strpos($table, '*PREFIX*') === 0) {
 			return $this->helper->quoteColumnName($table);
 		}
 
 		return $this->helper->quoteColumnName('*PREFIX*' . $table);
 	}
+
+	/**
+	 * Returns the table name quoted and with database prefix as needed by the implementation
```
Comment (rullzer) [inline lib/private/db/querybuilder/querybuilder.php]: s/table/column

File: lib/public/db/querybuilder/iquerybuilder.php [hunk 2]
```diff
@@ -796,4 +812,31 @@ public function createParameter($name);
...
+	 * Returns the table name quoted and with database prefix as needed by the implementation
+	 *
+	 * @param string $table
+	 * @return string
+	 * @since 9.0.0
+	 */
+	public function getTableName($table);
+
+	/**
+	 * Returns the table name quoted and with database prefix as needed by the implementation
```
Comment (rullzer) [inline lib/public/db/querybuilder/iquerybuilder.php]: s/table/column

**Comment by rullzer** [comment]
Nice additions! :+1:

File: lib/private/db/querybuilder/querybuilder.php (hunk 1 above)
Comment (MorrisJobke) [inline lib/private/db/querybuilder/querybuilder.php]: and without a database prefix ;)

File: lib/public/db/querybuilder/iquerybuilder.php (hunk 2 above)
Comment (MorrisJobke) [inline lib/public/db/querybuilder/iquerybuilder.php]: see above

**Comment by MorrisJobke** [comment]
Beside the PHPDoc comments :+1:

**Comment by nickvergessen** [comment]
Fixed le docs

**Comment by LukasReschke** [comment]
Failing on Oracle: https://github.com/owncloud/core/issues/21052

**Comment by LukasReschke** [comment]
Failing on PGSQL: https://github.com/owncloud/core/issues/21057

**Comment by LukasReschke** [comment]
Revert is at https://github.com/owncloud/core/pull/21058

**Pull request merged**